query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithoutFilter() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.id").build());
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQuerySingleDistinctSelection() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.setDistinctSelections(true).addSelection(createColumnExpression("Span.id"));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select distinct encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryMultipleDistinctSelection() {
+    Builder builder = QueryRequest.newBuilder();
+    builder
+        .setDistinctSelections(true)
+        .addSelection(createColumnExpression("Span.id"))
+        .addSelection(createColumnExpression("Span.displaySpanName"))
+        .addSelection(createColumnExpression("Span.serviceName"));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select distinct encode(span_id, 'hex'), span_name, service_name FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithStringFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.displaySpanName", "GET /login"));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_name = 'GET /login'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testSQLiWithStringValueFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(
+            createEqualsFilter("Span.displaySpanName", "GET /login' OR tenant_id = 'tenant2"));
```

Comment:
What is the difference with `testQueryWithStringFilter`? It compares entire string, right?

Reply:
Is this to do with PrepareStatement?

Reply:
Copied from pinot. Need to check this part.